### DISCUSSION TITLE
I need help regarding the billing voucher code

### DISCUSSION THREAD

**Discussion by secureweb16** [Kyon147/laravel-shopify#272] [Q&A]
Are voucher codes possible when booking a plan? If so, how does Shopify know which code works and what value it has? Please let me
know the solutions that are possible, If yes then please share it's related information. How it will done using this package.

I have tried to find the details, But the related information is not available.

Currently, I have simply passed the plan ID and how it will work with the discount code.

`<p><a href="{{ route('billing', ['plan' => 2, 'shop' => Auth::user()->name, 'host' =>  app('request')->input('host') ]) }}">Upgrade</a></p>`

**Comment by Kyon147** [comment]
If you want to offer discounts you can use the api to build a charge with a percentage discount.  https://shopify.dev/docs/api/admin-graphql/2024-01/mutations/appSubscriptionCreate#examples-Create_a_subscription_with_a_percentage_value_discount

I've not heard of way of applying voucher codes - so you'd probably have to build this yourself and then generate the correct charge through the api as shown above.
